flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/util/StateConfigUtil.java

```diff
@@ -46,10 +41,7 @@ public static StateTtlConfig createTtlConfig(long retentionTime) {
         }
     }
 
-    public static boolean isStateImmutableInStateBackend(KeyedStateBackend<?> stateBackend) {
-        // TODO: remove the hard code check once FLINK-21027 is supported
-        return (stateBackend instanceof AbstractKeyedStateBackend)
-                && ((AbstractKeyedStateBackend<?>) stateBackend)
-                        .isStateImmutableInStateBackend(CheckpointType.CHECKPOINT);
+    public static boolean isStateImmutableInStateBackend(KeyedStateBackend<?> keyedStateBackend) {
+        return keyedStateBackend.isStateImmutableInStateBackend(CheckpointType.CHECKPOINT);
```

Comment:
```suggestion
        // TODO: remove the hard code check once FLINK-21027 is supported
        return keyedStateBackend.isStateImmutableInStateBackend(CheckpointType.CHECKPOINT);
```
We need to keep this TODO comment since invocation of this `isStateImmutableInStateBackend` method should be replaced by the to-be-added `isStateKeyValueSerialized` method after FLINK-21027 is implemented.